### PR TITLE
Collab/ai studio

### PR DESCRIPTION
**Pull request by jhajduk-microsoft**
Adding ACR changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -470,7 +470,7 @@ You will need access to the prompt flow files for this experience, since we'll b
 
     The following code will create a folder named 'dist' with a Dockerfile and all the required flow files.
 
-TODO (P1 Jon): Stopped here. THere are two likely TODOs in the acr.bicep file to help make the following work.
+TODO (P1 Jen ): Stopped here. THere are two likely TODOs in the acr.bicep file to help make the following work.
```
Comment (ckittel) [inline README.md:473]: This is still part of your todo, I didn't see the `az acr build -t $FULL_IMAGE_NAME -r $NAME_OF_ACR .` command line change.  Don't you need to use the name of the agent as a parameter?

File: infra-as-code/bicep/modules/keyvaultRoleAssignment.bicep [hunk 2]
```diff
@@ -28,4 +28,5 @@ resource roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
     principalId: principalId
     principalType: 'ServicePrincipal'
   }
+
```
Comment (ckittel) [inline infra-as-code/bicep/modules/keyvaultRoleAssignment.bicep:31]: Please revert this change.

File: infra-as-code/bicep/acr.bicep [hunk 3]
```diff
@@ -56,20 +82,41 @@ resource acrResource 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
...
-  //  properties: {
-  //    os: 'Linux'
-  //    count: 1
-  //    virtualNetworkSubnetResourceId: // TODO
-  //  }
-  //}
+  // TODO (P2 - Jen): I believe the user is going to need AcrPush in order to put the new image into ACR.
 }
 
+resource acrBuildAgentNode 'Microsoft.ContainerRegistry/registries/agentPools@2019-06-01-preview' = {
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:90]: Please make a nested resource under the ACR resource.

File: infra-as-code/bicep/acr.bicep [hunk 4]
```diff
@@ -56,20 +82,41 @@ resource acrResource 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
...
+  name: acrBuildAgentNodeName
+  location: location
+  parent: acrResource
+  properties: {
+    count: 1
+    os: 'Linux'
+    tier: linuxNodeAgentTier
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:97]: Please hardcode the `tier`

File: infra-as-code/bicep/acr.bicep [hunk 5]
```diff
@@ -56,20 +82,41 @@ resource acrResource 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
...
+}
+
+resource acrPushRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(subscription().id, acrManagedIdentity.id, acrPushUserRole.id)
+  scope: acrResource
+  properties: {
+    roleDefinitionId: acrPushUserRole.id
+    principalId: acrManagedIdentity.properties.principalId
+  }
+}
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:109]: Are you sure this role assignment is needed?  Can you test without it once and see what happens?  Make sure the network for ACR has bypass enabled for `AzureServices`

File: infra-as-code/bicep/acr.bicep [hunk 6]
```diff
@@ -56,20 +82,41 @@ resource acrResource 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
...
+
+// ACR Pull Role Assignment
+resource acrPullRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(subscription().id, acrManagedIdentity.id, acrPullUserRole.id)
+  scope: acrResource
+  properties: {
+    roleDefinitionId: acrPullUserRole.id
+    principalId: acrManagedIdentity.properties.principalId
+  }
+}
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:119]: Can you let me know why it needs pull access in this scenario?

File: infra-as-code/bicep/acr.bicep [hunk 7]
```diff
@@ -13,16 +14,41 @@ param location string = resourceGroup().location
...
 param logWorkspaceName string
 
 // Variables
 var acrName = 'cr${baseName}'
+var acrManagedIdentityName = 'id-${acrName}'
 var acrPrivateEndpointName = 'pep-${acrName}'
 var acrDnsZoneName = 'privatelink${environment().suffixes.acrLoginServer}'
+var acrSubnetID = resourceId('Microsoft.Network/VirtualNetworks/subnets', vnetName, privateEndpointsSubnetName)
+
+// Built-in Azure RBAC role that is applied to a Azure Container Registry to grant with secrets content read privileges. Granted to both Key Vault and our workload's identity.
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:36]: This comment doesn't seem right.

File: infra-as-code/bicep/acr.bicep [hunk 8]
```diff
@@ -13,16 +14,41 @@ param location string = resourceGroup().location
...
+resource acrPushUserRole 'Microsoft.Authorization/roleDefinitions@2022-04-01' existing = {
+  name: '8311e382-0749-4cb8-b61a-304f252e45ec'
+  scope: subscription()
+}
+
+// Built-in Azure RBAC role that is applied to a Azure Container Registry to grant with secrets content read privileges. Granted to both Key Vault and our workload's identity.
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:42]: This comment doesn't seem right.

File: infra-as-code/bicep/acr.bicep [hunk 9]
```diff
@@ -13,16 +14,41 @@ param location string = resourceGroup().location
...
+resource acrPullUserRole 'Microsoft.Authorization/roleDefinitions@2022-04-01' existing = {
+  name: '7f951dda-4ed3-4680-a7ca-43fe172d538d'
+  scope: subscription()
+}
+
+resource acrManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:48]: Where is this identity used?

File: infra-as-code/bicep/acr.bicep [hunk 10]
```diff
@@ -13,16 +14,41 @@ param location string = resourceGroup().location
...
+
 @description('The name of the workload\'s existing Log Analytics workspace.')
 param logWorkspaceName string
 
 // Variables
 var acrName = 'cr${baseName}'
+var acrManagedIdentityName = 'id-${acrName}'
 var acrPrivateEndpointName = 'pep-${acrName}'
 var acrDnsZoneName = 'privatelink${environment().suffixes.acrLoginServer}'
+var acrSubnetID = resourceId('Microsoft.Network/VirtualNetworks/subnets', vnetName, privateEndpointsSubnetName)
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:34]: Please move this to be an `existing` resource, like you see on line `57`

File: infra-as-code/bicep/acr.bicep [hunk 11]
```diff
@@ -56,20 +82,41 @@ resource acrResource 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
...
-  //    os: 'Linux'
-  //    count: 1
-  //    virtualNetworkSubnetResourceId: // TODO
-  //  }
-  //}
+  // TODO (P2 - Jen): I believe the user is going to need AcrPush in order to put the new image into ACR.
 }
 
+resource acrBuildAgentNode 'Microsoft.ContainerRegistry/registries/agentPools@2019-06-01-preview' = {
+  name: acrBuildAgentNodeName
```
Comment (ckittel) [inline infra-as-code/bicep/acr.bicep:91]: Please hardcode this `name`

**Review by ckittel — CHANGES_REQUESTED** [review]

File: infra-as-code/bicep/acr.bicep (hunk 10 above)
Comment (ckittel) [reply infra-as-code/bicep/acr.bicep:34]: Also, this isn't the correct subnet.

Also, this subnet needs firewall rules adjusted per <https://learn.microsoft.com/azure/container-registry/tasks-agent-pools#create-pool-in-a-virtual-network>

File: infra-as-code/bicep/acr.bicep (hunk 6 above)
Comment (jhajduk-microsoft) [reply infra-as-code/bicep/acr.bicep:119]: I am not super familiar with the scenario - only wrapping up this one task. Ill remove it

File: infra-as-code/bicep/acr.bicep (hunk 11 above)
Comment (jhajduk-microsoft) [reply infra-as-code/bicep/acr.bicep:91]: Do you want me to hard code a GUID? The "name" has to be a guid

Comment (ckittel) [reply infra-as-code/bicep/acr.bicep:91]: Nah, just a regular alpha resource name is fine. I already have a name for this in this branch, so let's stick with that unless you have something different that is better.
